src/main/java/org/jabref/logic/importer/fetcher/Medra.java

```diff
@@ -0,0 +1,175 @@
+            entry.setType(convertType(item.getString("type")));
+            entry.setField(StandardField.TITLE, item.getString("title"));
+            entry.setField(StandardField.AUTHOR, toAuthors(item.optJSONArray("author")));
+            entry.setField(StandardField.YEAR,
+                           Optional.ofNullable(item.optJSONObject("issued"))
+                                   .map(array -> array.optJSONArray("date-parts"))
+                                   .map(array -> array.optJSONArray(0))
+                                   .map(array -> array.optInt(0))
+                                   .map(year -> Integer.toString(year)).orElse(""));
+            entry.setField(StandardField.DOI, item.getString("DOI"));
+            entry.setField(StandardField.PAGES, item.optString("page"));
+            entry.setField(StandardField.ISSN, item.optString("ISSN"));
+            entry.setField(StandardField.JOURNAL, item.optString("container-title"));
+            entry.setField(StandardField.PUBLISHER, item.optString("publisher"));
+            entry.setField(StandardField.URL, item.optString("URL"));
+            entry.setField(StandardField.VOLUME, item.optString("volume"));
+            return entry;
+        } catch (JSONException exception) {
+            throw new ParseException("mEdRA API JSON format has changed", exception);
+        }
+    }
+
+    private EntryType convertType(String type) {
+        switch (type) {
+            case "article-journal":
+                return StandardEntryType.Article;
+            default:
+                return StandardEntryType.Misc;
+        }
+    }
+
+    private String toAuthors(JSONArray authors) {
+        if (authors == null) {
+            return "";
+        }
+
+        // input: list of {"literal":"A."}
+        AuthorList authorsParsed = new AuthorList();
+        String name = "";
+
+        for (int i = 0; i < authors.length(); i++) {
+            JSONObject author = authors.getJSONObject(i);
+            if (author.has("literal")) {
+                name = author.optString("literal", "");
+            } else {
+                name = author.optString("family", "") + " " + author.optString("given", "");
+            }
+
+            authorsParsed.addAuthor(
+                                    name,
+                                    "",
+                                    "",
+                                    "",
+                                    "");
+
+        }
+        return authorsParsed.getAsFirstLastNamesWithAnd();
+    }
+
+    @Override
+    public Optional<BibEntry> performSearchById(String identifier) throws FetcherException {
+
+        try (InputStream stream = getUrlDownload(identifier).asInputStream();
+             PushbackInputStream pushbackInputStream = new PushbackInputStream(stream)) {
+
+            List<BibEntry> fetchedEntries = new ArrayList<>();
+
+            // check if there is anything to read since mEDRA '404 not found' returns nothing
```

Comment:
Ah, you mean HTTP 404 returns an empty response.

In line 259 of URLDownload is is implemented exactly as that.

Why do the other fetchers cope well with that and here you have to do some special tweaks?

Maybe, the response from the mEDRA thing is different?

Reply:
You're right, URLDownload returns empty stream but I thought somewhere I had to check whether it's empty or not before trying to parse it to JSON.
Crossref serach by ID doesn't cope well with that, for example, because there's not empty stream check. I added a test for valid DOI returning nothing to CrossRefTest and the test failed.
If you think this could be the case I could implement just that check on performSearchById of Crossref as well, otherwise i just commit my changes.

![2020-07-04_18h11_15](https://user-images.githubusercontent.com/59294994/86517322-5d35d600-be28-11ea-8d16-8298808ba389.png)

Reply:
I think it makes sense to add that to CrossRef as well

Reply:
I think, this is fixed in https://github.com/mind000/jabref/pull/1/, too.